.github/workflows/coverage_runner.yml

```diff
@@ -0,0 +1,109 @@
+name: Run tests, measure coverage and upload results to codecov
+on:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: Enter PR number coming from community to run coverage on it. Do not run for PR coming from hazelcast organization, they will be run automatically.
+        required: true
+  push:
+    branches:
+      - master
+      - '[45].*.z'
+  pull_request_target:
+    branches:
+      - master
+      - '[45].*.z'
+    types:
+      [opened]
+jobs:
+  check_for_membership:
+    runs-on: ubuntu-latest
+    name: Check membership of given user  
+    outputs:
+      check-result: ${{ steps.composite.outputs.check-result }}
+    steps: 
+      - name: Action for membership check
+        id: composite
+        uses: hazelcast/hazelcast-tpm/membership@main
+        with:
+          organization-name: 'hazelcast'
+          member-name: ${{ github.event.pull_request.head.repo.owner.login }}
+          token: ${{ secrets.PAT }}
+          
+  run-tests:
+    runs-on: ${{ matrix.os }}
+    needs: [check_for_membership]
+    if: github.event_name == 'push' || needs.check_for_membership.outputs.check-result == 'true' || github.event_name == 'workflow_dispatch'
+    name: Run tests with Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.10' ]
+        os: [ ubuntu-latest, windows-latest ]
+      fail-fast: false
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+          
+      - name: Checkout code for push event
+        if: github.event_name == 'push'
+        uses: actions/checkout@v2
+
+      - name: Checkout code for PR
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          
+      - name: Checkout repository to fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v2
+        
+      - name: Fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git fetch origin pull/${{github.event.inputs.pr_number }}/head:temp_branch
```

Comment:
We can add a comment about `pull/${{github.event.inputs.pr_number }}/head`. It is not something widely known. I had to google it and found [this](https://stackoverflow.com/questions/63594658/git-refs-merge-vs-head-in-pull-request). I would add a comment like this:

"pull/<number>/head" is an automatically generated reference to HEAD commit of the PR.


Also can you rename `temp_branch` to pr_head or something like that to make it more clear?

Reply:
Fixed at https://github.com/hazelcast/hazelcast-python-client/commit/7cb06a7488d96c1f42d758b6ac12900be23a3051